laravel-app/app/Http/Controllers/PostController.php

```diff
@@ -3,37 +3,62 @@
 namespace App\Http\Controllers;
 
 use App\Http\Requests\UserPostRequest;
+use App\Models\UserPost;
+use Illuminate\Http\RedirectResponse;
 use Illuminate\View\View;
+use App\Models\User;
 
 class PostController extends Controller
 {
+    /* 
+    * View all instances of posts in home page.
+    */
+    public function index() : View
+    {
+        /** @var User $user */
+        $user = auth()->user();
+        $posts = $user->posts->orderByDsc('created_at')->get();
+    
+        return view('home', compact('user', 'posts'));
+    }    
+
     /**
-     * Show login form page.
-     *
-     * @return View
+     * Show create post page.
      */
-    public function index()
+    public function create() : View
     {
-        return view('post.index');
+        $user = auth()->user();
+        return view('post.create', compact('user'));
     }
 
     /**
-     *
-     * @param UserPostRequest $request
-     * @return View
+     * Save post content as new post.
      */
-    public function store(UserPostRequest $request)
+    public function store(UserPostRequest $request) : RedirectResponse
     {
-        $validatedData = $request->validated();
+        /** @var User $user */
         $user = auth()->user();
+        $validatedData = $request->validated();
 
-        /**
-         * @var \App\Models\User $user
-         */
-        $user->posts()->create([
+        /** @var \App\Models\User $user */
+        $post = $user->posts()->create([
             'content' => $validatedData['content'],
         ]);
 
-        return view('post.display');
+        if ($post !== null) {
+            $success = ['success' => 'Post created successfully'];
+            
+            return redirect()->route('post.show', ['post' => $post])->with($success);
+        } else {
+            return back()->with('error', 'Failed to create post');
+        }
```

Comment:
Prevent else in the return statement.

https://thomas-sanctorum.medium.com/how-to-simply-avoid-else-in-php-and-others-programming-language-ccd6a07c4e1